spec.bs

```diff
@@ -1717,3 +1755,160 @@ specification is printed below:
+  1. If |container| is a [=fenced navigable container=], then let <var ignore>policy</var> be the
+     result of running
+     [=Create a Permissions Policy for a fenced navigable=] given |container|, |origin|, and
+     |effective permissions|.
+  
+     Otherwise, let <var ignore>policy</var> be the result of running [=Create a Permissions Policy
+     for a navigable=] given |container| and |origin|.
+</div>
+
+<div algorithm=shared-document-creation-changes>
+  Modify the [=create and initialize a Document object=] algorithm. Rewrite step 3 to read:
+
+  3. Let <var ignore>permissionsPolicy</var> be the result of [=Create a Permissions Policy for a
+     navigable from response|creating a permissions policy from a response=] given
+     |navigationParams|'s [=navigable=]'s [=navigable container|container=], |navigationParams|'s
+     [=navigation params/origin=], |navigationParams|'s [=navigation params/response=], and
+     |navigationParams|'s [=navigable=]'s [=fenced frame config instance=]'s [=fenced frame config
+     instance/effective permissions=].
+</div>
+
+<div algorithm=new-browsing-context-changes>
+  Modify the [=create a new browsing context and document=] algorithm. Rewrite step 7 to read:
+
+  7. If <var ignore>navigationParams</var>'s [=navigable=]'s [=navigable/traversable navigable=] is
+     a [=fenced navigable container/fenced navigable=], let |permissionsPolicy| be the result of
+     [=Create a Permissions Policy for a fenced navigable|creating a permissions policy for a fenced
+     navigable=] given |embedder|, |origin|, and null.
+
+     Otherwise, let |permissionsPolicy| be the result of [=Create a Permissions Policy for a
+     navigable|creating a permissions policy=] given |embedder| and
+     |origin|.
+
+  Note: This change is made in addition to the changes to [=create a new browsing context and
+  document=] outlined in [[#creating-browsing-contexts-patch]].
+</div>
+
+<div algorithm=attempt-populate-history-patches>
+  Modify [[HTML]]'s [=attempt to populate the history entry's document=] algorithm. Add a step
+  before the step inside the [=queue a task|queued task=] starting with "If
+  |failure| is true, then:" that reads:
+
+  8. Otherwise, if the result of [=Should navigation response to navigation request be blocked by
+    Permissions Policy?=] given <var ignore>navigationParams</var>'s [=response=], and <var
+    ignore>navigable</var> is "`Blocked`", then set |failure| to true.
+</div>
+
+<div algorithm=permissions-policy-block-request>
+  Create a new algorithm called <dfn>Should navigation response to navigation request be blocked by
+  Permissions Policy?</dfn> in [[!HTML]].
+
+  Given a [=response=] (|response|) and a [=navigable=] (|navigable|), this algorithm returns
+  `Blocked` or `Allowed`:
+
+  1. If |navigable| is not a [=fenced navigable container/fenced navigable=], then return `Allowed`.
+
+  2. Let |origin| be |response|'s [=response/url=]'s [=url/origin=].
```

Comment:
We shouldn't get the response's origin, but rather the navigation params's origin (https://html.spec.whatwg.org/C#navigation-params-origin), as this will be the actual origin used for the document.